### ISSUE TITLE
SkywalkingAgent-5-GRPCChannelManager-0 GRPCChannelManager : Create channel to localhost:11800-noverify fail

### ISSUE THREAD

**Issue by nhx7103972** [apache/skywalking#6848]
SkyWalking8.4.0 
OS：window10
JDK：1.8.202
springcloud gateway ：2.2.5.RELEASE
springcloud : Hoxton.SR8
spring-cloud-alibaba-dependencies : 2.2.4.RELEASE
springboot : 2.3.2.RELEASE
elasticsearch : 7.11.2
agent plugin : apm-spring-cloud-gateway-2.1.x-plugin-8.4.0.jar  apm-spring-webflux-5.x-plugin-8.4.0.jar

gateway start，agent\logs\skywalking-api.log

ERROR 2021-04-27 10:08:17:328 SkywalkingAgent-5-GRPCChannelManager-0 GRPCChannelManager : Create channel to 127.0.0.1:11800-noverify fail. 
java.lang.NumberFormatException: For input string: "11800-noverify"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at org.apache.skywalking.apm.agent.core.remote.GRPCChannelManager.run(GRPCChannelManager.java:133)
	at org.apache.skywalking.apm.util.RunnableWithExceptionProtection.run(RunnableWithExceptionProtection.java:33)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by wu-sheng** [comment]
Isn't the error very clear? `11800-noverify` is an illegal port number. Why asking us?

**Comment by nhx7103972** [comment]
I didn't change the port number

apache-skywalking-apm-es7-8.4.0\apache-skywalking-apm-bin-es7\config\application.yml

gRPCHost: ${SW_CORE_GRPC_HOST:0.0.0.0}
gRPCPort: ${SW_CORE_GRPC_PORT:11800}

**Comment by wu-sheng** [comment]
This is your env, and the log is very clear. I think this is nothing we can do, it is on your hands.

**Comment by nhx7103972** [comment]
Thank you for your help。

When I start gateway , add vm arguments .

-javaagent:E:\tool\apache-skywalking-apm-es7-8.4.0\apache-skywalking-apm-bin-es7\agent\skywalking-agent.jar -Dskywalking.agent.service_name=gateway -Dskywalking.collector.backend_service=127.0.0.1:11800

I cancel this part of the arguments 
-Dskywalking.collector.backend_service=127.0.0.1:11800

Start successfully。

**Comment by tubushitu** [comment]
ERROR 2023-01-18 20:07:55:563 SkywalkingAgent-5-GRPCChannelManager-0 GRPCChannelManager : Create channel to ${SKYWALKING_SERVER} fail.
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
  at org.apache.skywalking.apm.agent.core.remote.GRPCChannelManager.run(GRPCChannelManager.java:133)
  at org.apache.skywalking.apm.util.RunnableWithExceptionProtection.run(RunnableWithExceptionProtection.java:33)
  at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
  at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
  at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
  at java.base/java.lang.Thread.run(Unknown Source)






I have encountered the same problem as yours, but my skywalking-Dskywalking.collector.backend_ Service=${SKYWALKING_SERVER} I passed it in the form of variables. The agent is in the container. I made the basic image of the docker
